internal/service/cloudbackupsnapshotrestorejob/data_source_cloud_backup_snapshot_restore_jobs.go

```diff
@@ -140,29 +137,30 @@ func dataSourcePluralRead(ctx context.Context, d *schema.ResourceData, meta any)
 	return nil
 }
 
-func flattenCloudProviderSnapshotRestoreJobs(cloudProviderSnapshotRestoreJobs []*matlas.CloudProviderSnapshotRestoreJob) []map[string]any {
+func flattenCloudProviderSnapshotRestoreJobs(cloudProviderSnapshotRestoreJobs []admin.DiskBackupSnapshotRestoreJob) []map[string]any {
 	var results []map[string]any
 
 	if len(cloudProviderSnapshotRestoreJobs) > 0 {
 		results = make([]map[string]any, len(cloudProviderSnapshotRestoreJobs))
 
-		for k, cloudProviderSnapshotRestoreJob := range cloudProviderSnapshotRestoreJobs {
+		for k := range cloudProviderSnapshotRestoreJobs {
+			cloudProviderSnapshotRestoreJob := cloudProviderSnapshotRestoreJobs[k]
 			results[k] = map[string]any{
-				"id":                        cloudProviderSnapshotRestoreJob.ID,
-				"cancelled":                 cloudProviderSnapshotRestoreJob.Cancelled,
-				"created_at":                cloudProviderSnapshotRestoreJob.CreatedAt,
-				"delivery_type":             cloudProviderSnapshotRestoreJob.DeliveryType,
-				"delivery_url":              cloudProviderSnapshotRestoreJob.DeliveryURL,
-				"expired":                   cloudProviderSnapshotRestoreJob.Expired,
-				"expires_at":                cloudProviderSnapshotRestoreJob.ExpiresAt,
-				"finished_at":               cloudProviderSnapshotRestoreJob.FinishedAt,
-				"snapshot_id":               cloudProviderSnapshotRestoreJob.SnapshotID,
-				"target_project_id":         cloudProviderSnapshotRestoreJob.TargetGroupID,
-				"target_cluster_name":       cloudProviderSnapshotRestoreJob.TargetClusterName,
-				"timestamp":                 cloudProviderSnapshotRestoreJob.Timestamp,
-				"oplog_ts":                  cloudProviderSnapshotRestoreJob.OplogTs,
-				"point_in_time_utc_seconds": cloudProviderSnapshotRestoreJob.PointInTimeUTCSeconds,
-				"oplog_inc":                 cloudProviderSnapshotRestoreJob.OplogInc,
+				"id":        cloudProviderSnapshotRestoreJob.GetId(),
```

Comment:
I used the getter methods here. As the legacy SDK doesn't use pointers, however, using time pointer conversion for the timestamps as the default timestamp value is probably unexpected.